local-cli/runWindows/utils/deploy.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const child_process = require('child_process');
+const spawn = child_process.spawn, execSync = child_process.execSync;
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+const chalk = require('chalk');
+const glob = require('glob');
+const parse = require('xml-parser');
+const WinAppDeployTool = require('./winappdeploytool');
+
+function getAppPackage(options) {
+  return glob.sync(path.join(options.root, 'windows/*/AppPackages/*'))[0];
+}
+
+function getWindowsStoreAppUtils() {
+  const appStorePath = path.join(__dirname, 'WindowsStoreAppUtils.ps1');
+  execSync(`powershell Unblock-File "${appStorePath}"`);
+  return appStorePath;
+}
+
+function getAppxManifest(options) {
+  const appxPath = glob.sync(path.join(options.root, 'windows/*/bin/*/*/AppxManifest.xml'))[0];
+  return parse(fs.readFileSync(appxPath, 'utf8'));
+}
+
+function handleResponseError(e) {
+  if (e.message.indexOf('Error code -2146233088')) {
+    throw new Error(`No Windows Mobile device was detected: ${e.message}`);
+  } else {
+    throw new Error(`Unexpected error deploying app: ${e.message}`);
+  }
+}
+
+// Errors: 0x80073d10 - bad architecture
+function deployToDevice(options) {
+  const appPackageFolder = getAppPackage(options);
+
+  const deployTarget = options.target ? options.target : (options.emulator ? 'emulator' : 'device');
+  const deployTool = new WinAppDeployTool();
+  const appxManifest = getAppxManifest(options);
+  const identity = appxManifest.root.children.filter(function (x) { return x.name === 'mp:PhoneIdentity'; })[0];
+  const appName = identity.attributes.PhoneProductId;
+
+  let device;
+  try {
+    device = deployTool.findDevice(deployTarget);
+  } catch (e) {
+    console.log(chalk.red(e.message));
+    return;
+  }
+
+  return new Promise(resolve => {
+    try {
+      deployTool.uninstallAppPackage(appName, device);
+    } catch (e) {
+      console.log(chalk.yellow('Failed to uninstall app from ' + device.name));
+    }
+
+    const appxFile = glob.sync(path.join(appPackageFolder, '*.appx'))[0];
+    try {
+      console.log(chalk.white(deployTool.installAppPackage(appxFile, device, true, false)));
+      resolve();
```

Comment:
Had too many issues with things running out of order so I moved all things back to sync inside a Promise